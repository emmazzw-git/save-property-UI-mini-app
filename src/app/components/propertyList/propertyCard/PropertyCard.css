.propertyCardContainer {
    display: flex;
    flex-direction: column;
    display: -webkit-flex; /* Safari */
    -webkit-flex-direction: row-reverse;/* Safari 6.1+ */
    width: 160px;
    overflow: hidden;
    border-radius: 5px;
    border: 1px solid darkgrey;
    margin: 10px 0 20px;
    position: relative;
}

.propertyCardHeader {
   text-align: left;
   padding: 5px 0;
}

.logo {
   width: 50px;
}

.propertyImage {
    width: 158px;
    height: 100px;
}

.propertySelectionOptionButton {
    position: absolute;
    bottom: 1px;
    right: 10px;
    font-size: 0.9em;
    border-radius: 10px;
    width: 140px
}

.addProperty {
    color: green;
    border: 2px solid green;
}

.removeProperty {
    color: red;
    border: 2px solid red;
}