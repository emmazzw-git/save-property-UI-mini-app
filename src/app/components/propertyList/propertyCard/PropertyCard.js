import React, { Component } from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'

import './PropertyCard.css'

export class PropertyCard extends Component {
  constructor (props) {
    super(props)
    this.state = {
      isVisible: false
    }
    this.handleClick = this.handleClick.bind(this)
    this.mouseEnter = this.mouseEnter.bind(this)
    this.mouseLeave = this.mouseLeave.bind(this)
  }

  mouseEnter () {
    this.setState({isVisible: true})
  }

  mouseLeave () {
    this.setState({isVisible: false})
  }

  handleClick () {
    const { callBackFn, propertyDetails } = this.props
    callBackFn(propertyDetails)
  }

  render () {
    const {isVisible} = this.state
    const { id, price, mainImage, agency } = this.props.propertyDetails
    const { buttonText, buttonClassName } = this.props
    const { brandingColors, logo } = agency
    const buttonStyle = classNames(
      'propertySelectionOptionButton',
      `${buttonClassName}`
    )
    return (
      <div className='propertyCardContainer'
        onMouseEnter={this.mouseEnter}
        onMouseLeave={this.mouseLeave}
        key={id}>
        <div className='propertyCardHeader' style={{ backgroundColor: brandingColors.primary }}>
          <img className='logo' src={logo} alt='logoImage' />
        </div>
        <div><img className='propertyImage' src={mainImage} alt='propertyImage' /></div>
        <div>{price}</div>
        {
          isVisible ? (
            <button className={buttonStyle} onClick={this.handleClick}>
              {buttonText}
            </button>) : null
        }
      </div>
    )
  }
}

PropertyCard.propTypes = {
  propertyDetails: PropTypes.shape({
    id: PropTypes.string,
    price: PropTypes.string,
    mainImage: PropTypes.string,
    agency: PropTypes.shape({
      brandingColors: PropTypes.shape({
        primary: PropTypes.string
      }),
      logo: PropTypes.string
    })
  }),
  buttonClassName: PropTypes.string,
  callBackFn: PropTypes.func
}

export default PropertyCard
