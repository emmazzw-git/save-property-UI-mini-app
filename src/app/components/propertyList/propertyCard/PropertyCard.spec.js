import React from 'react'
import { shallow } from 'enzyme'
import PropertyCard from './PropertyCard'

describe('PropertyCard', () => {
  const props = {
    propertyDetails: {
      price: '$726,500',
      agency: {
        brandingColors: {
          primary: '#ffe512'
        },
        logo: 'http://i1.au.reastatic.net/agencylogo/XRWXMT/12/20120927204448.gif'
      },
      id: '1',
      mainImage: 'http://i2.au.reastatic.net/640x480/20bfc8668a30e8cabf045a1cd54814a9042fc715a8be683ba196898333d68cec/main.jpg'
    },
    buttonClassName: 'addProperty',
    callBackFn: jest.fn()
  }

  test('snapshot test', () => {
    const component = shallow(<PropertyCard {...props} />)
    expect(component).toMatchSnapshot()
  })

  test('should set isVisible state to true when the mouse is hovering the property card', () => {
    const component = shallow(<PropertyCard {...props} />)
    component.instance().mouseEnter()
    expect(component.state().isVisible).toEqual(true)
  })

  test('should set isVisible state to false when the mouse loses hover of the property card', () => {
    const component = shallow(<PropertyCard {...props} />)
    component.instance().mouseLeave()
    expect(component.state().isVisible).toEqual(false)
  })

  test('should call props.callBackFn when click on the button', () => {
    const component = shallow(<PropertyCard {...props} />)
    component.instance().handleClick()
    expect(props.callBackFn).toHaveBeenCalledWith(props.propertyDetails)
  })
})
