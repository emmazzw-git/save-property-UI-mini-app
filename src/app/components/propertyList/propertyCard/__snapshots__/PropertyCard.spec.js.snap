// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PropertyCard snapshot test 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PropertyCard
    buttonClassName="addProperty"
    callBackFn={[Function]}
    propertyDetails={
        Object {
            "agency": Object {
              "brandingColors": Object {
                "primary": "#ffe512",
              },
              "logo": "http://i1.au.reastatic.net/agencylogo/XRWXMT/12/20120927204448.gif",
            },
            "id": "1",
            "mainImage": "http://i2.au.reastatic.net/640x480/20bfc8668a30e8cabf045a1cd54814a9042fc715a8be683ba196898333d68cec/main.jpg",
            "price": "$726,500",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": "1",
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="propertyCardHeader"
          style={
                    Object {
                              "backgroundColor": "#ffe512",
                            }
          }
>
          <img
                    alt="logoImage"
                    className="logo"
                    src="http://i1.au.reastatic.net/agencylogo/XRWXMT/12/20120927204448.gif"
          />
</div>,
        <div>
          <img
                    alt="propertyImage"
                    className="propertyImage"
                    src="http://i2.au.reastatic.net/640x480/20bfc8668a30e8cabf045a1cd54814a9042fc715a8be683ba196898333d68cec/main.jpg"
          />
</div>,
        <div>
          $726,500
</div>,
        null,
      ],
      "className": "propertyCardContainer",
      "onMouseEnter": [Function],
      "onMouseLeave": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <img
            alt="logoImage"
            className="logo"
            src="http://i1.au.reastatic.net/agencylogo/XRWXMT/12/20120927204448.gif"
/>,
          "className": "propertyCardHeader",
          "style": Object {
            "backgroundColor": "#ffe512",
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "logoImage",
            "className": "logo",
            "src": "http://i1.au.reastatic.net/agencylogo/XRWXMT/12/20120927204448.gif",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <img
            alt="propertyImage"
            className="propertyImage"
            src="http://i2.au.reastatic.net/640x480/20bfc8668a30e8cabf045a1cd54814a9042fc715a8be683ba196898333d68cec/main.jpg"
/>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "propertyImage",
            "className": "propertyImage",
            "src": "http://i2.au.reastatic.net/640x480/20bfc8668a30e8cabf045a1cd54814a9042fc715a8be683ba196898333d68cec/main.jpg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "$726,500",
        },
        "ref": null,
        "rendered": "$726,500",
        "type": "div",
      },
      null,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": "1",
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="propertyCardHeader"
            style={
                        Object {
                                    "backgroundColor": "#ffe512",
                                  }
            }
>
            <img
                        alt="logoImage"
                        className="logo"
                        src="http://i1.au.reastatic.net/agencylogo/XRWXMT/12/20120927204448.gif"
            />
</div>,
          <div>
            <img
                        alt="propertyImage"
                        className="propertyImage"
                        src="http://i2.au.reastatic.net/640x480/20bfc8668a30e8cabf045a1cd54814a9042fc715a8be683ba196898333d68cec/main.jpg"
            />
</div>,
          <div>
            $726,500
</div>,
          null,
        ],
        "className": "propertyCardContainer",
        "onMouseEnter": [Function],
        "onMouseLeave": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              alt="logoImage"
              className="logo"
              src="http://i1.au.reastatic.net/agencylogo/XRWXMT/12/20120927204448.gif"
/>,
            "className": "propertyCardHeader",
            "style": Object {
              "backgroundColor": "#ffe512",
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "logoImage",
              "className": "logo",
              "src": "http://i1.au.reastatic.net/agencylogo/XRWXMT/12/20120927204448.gif",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              alt="propertyImage"
              className="propertyImage"
              src="http://i2.au.reastatic.net/640x480/20bfc8668a30e8cabf045a1cd54814a9042fc715a8be683ba196898333d68cec/main.jpg"
/>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "propertyImage",
              "className": "propertyImage",
              "src": "http://i2.au.reastatic.net/640x480/20bfc8668a30e8cabf045a1cd54814a9042fc715a8be683ba196898333d68cec/main.jpg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "$726,500",
          },
          "ref": null,
          "rendered": "$726,500",
          "type": "div",
        },
        null,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
